jcommune-view/jcommune-web-view/src/test/java/org/jtalks/jcommune/test/AdministrationControllerTest.groovy

```diff
@@ -109,12 +107,117 @@ class AdministrationControllerTest extends Specification {
+            def session = users.signInAsAdmin(forum)
+            def savedGroupId = groupsService.save(Groups.random())
+            def savedGroupName = groupsService.getById(savedGroupId).name
+        when: 'Admin attempts to edit group'
+            def dto = new GroupAdministrationDto()
+            dto.setId(savedGroupId)
+            dto.setName(alphanumeric(GROUP_NAME_MAX_LENGTH))
+            dto.setDescription(alphanumeric(GROUP_DESCRIPTION_MAX_LENGTH))
+            groups.edit(dto, session)
+        then:
+            groups.assertDoesNotExistWithName(savedGroupName)
+    }
+
+    def 'must not be able to set invalid name or description'() {
+        given: 'Admin signed in and forum has some existing group'
+        def session = users.signInAsAdmin(forum)
+        def savedGroupId = groupsService.save(Groups.random())
+        when: 'Admin attempts to edit group'
+        def dto = new GroupAdministrationDto()
+        dto.setId(groupId.toInteger())
+        dto.setName(groupName)
+        dto.setDescription(groupDescription)
+        groups.edit(dto, session)
+        then: 'Validation error occurs'
+        def e = thrown(ValidationException)
+        [errorMessage] == e.defaultErrorMessages
+        and: 'Group is not edited'
+        groups.assertDoesNotExistWithName(dto.name)
+        where:
+        groupId                 |groupName           | groupDescription    |errorMessage                                                    | caseName
+        savedGroupId.toString() |''                  | alphanumeric(0,255) |"Group name length must be between 1 and 100 characters"        | 'Group name is empty'
+        savedGroupId.toString() |'     '             | alphanumeric(0,255) |"Group name length must be between 1 and 100 characters"        | 'Group name contains only spaces'
+        savedGroupId.toString() |alphanumeric(101)   | alphanumeric(0,255) |"Group name length must be between 1 and 100 characters"        | 'Group name too long'
+        savedGroupId.toString() |alphanumeric(1,100) | alphanumeric(256)   |"Group description length must be between 0 and 255 characters" | 'Group description too long'
+    }
+
+    def 'must not be able to edit non editable group'() {
+        given: 'Admin signed in and forum has some existing group'
+            def session = users.signInAsAdmin(forum)
+        when: 'Admin attempts to edit group'
+            def dto = new GroupAdministrationDto()
+            dto.setId(groupsService.getIdByName(groupName.name))
+            dto.setName(alphanumeric(GROUP_NAME_MAX_LENGTH))
+            dto.setDescription(alphanumeric(GROUP_DESCRIPTION_MAX_LENGTH))
+            groups.edit(dto, session)
+        then:
+            thrown(WrongResponseException)
+        and: 'Group is not edited'
+            groups.assertDoesNotExistWithName(dto.name)
+        where:
+            groupName   | caseName
+            ADMIN       | 'Group name is ' + ADMIN.name
```

Comment:
Can be changed to `groupName << [ADMIN, USER, BANNED_USER]`. Not sure if the case name is really necessary here.